### PR TITLE
[SR GeekCamp] Improve schema scan performance #4028

### PR DESCRIPTION
**Pull request by ylcq**
Improve point: 
1. add limit parameter;
2. schema_scan_node add db & table parameter when user set db & table in where condition (for binary predicate)

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 14 / 20 (70.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|10|10|100.00%|
|:large_blue_circle:|com/starrocks/planner/SchemaScanNode.java|4|4|100.00%|

File: be/src/exec/vectorized/schema_scanner.h [hunk 1]
```diff
@@ -31,6 +31,7 @@ struct SchemaScannerParam {
     const std::string* ip{nullptr};                   // frontend ip
     int32_t port{0};                                  // frontend thrift port
     int64_t thread_id = 0;
+    int32_t limit{0};
```
Comment (imay) [inline be/src/exec/vectorized/schema_scanner.h:34]: int32_t is not enough, I think int64_t is OK

File: gensrc/thrift/FrontendService.thrift [hunk 2]
```diff
@@ -74,6 +74,7 @@ struct TDescribeTableParams {
   3: optional string user   // deprecated
   4: optional string user_ip    // deprecated
   5: optional Types.TUserIdentity current_user_ident // to replace the user and user ip
+  6: optional i32 limit
```
Comment (imay) [inline gensrc/thrift/FrontendService.thrift:77]: `limit` should not be add in TDescribeTableParams
I think TPlanNode contains the limit, you can refer to be/src/exec/vectorized/olap_scan_node.cpp to see how to get the limit.

Comment (kangkaisen) [reply gensrc/thrift/FrontendService.thrift:77]: `_limit` is the ExecNode field. so you could directly access limit in SchemaScanNode.

please refer to 

```
    _num_rows_returned += chunk_dst->num_rows();
    if (reached_limit()) {
        int64_t num_rows_over = _num_rows_returned - _limit;
        chunk_dst->set_num_rows(chunk_dst->num_rows() - num_rows_over);
        COUNTER_SET(_rows_returned_counter, _limit);
    } else {
        COUNTER_SET(_rows_returned_counter, _num_rows_returned);
    }
    *chunk = std::move(chunk_dst);
```

File: gensrc/thrift/FrontendService.thrift [hunk 3]
```diff
@@ -310,6 +311,7 @@ struct TGetTablesParams {
   4: optional string user_ip    // deprecated
   5: optional Types.TUserIdentity current_user_ident // to replace the user and user ip
   20: optional Types.TTableType type // getting a certain type of tables
+  21: optional i32 limit
```
Comment (kangkaisen) [inline gensrc/thrift/FrontendService.thrift:314]: needn't

File: gensrc/thrift/FrontendService.thrift (hunk 2 above)
Comment (ylcq) [reply gensrc/thrift/FrontendService.thrift:77]: The limit parameter is passed from BE to FE, then which can reduce the amount of data transmitted from FE to BE.

File: be/src/exec/vectorized/schema_scanner.h (hunk 1 above)
Comment (ylcq) [reply be/src/exec/vectorized/schema_scanner.h:34]: ok, I will modify it later

**Comment by kangkaisen** [comment]
@ylcq Hi, you need to sign the CLA

**Comment by ylcq** [comment]
> @ylcq Hi, you need to sign the CLA

I've signed in, but it still says I'm not signed yet.

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 14 / 20 (70.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|10|10|100.00%|
|:large_blue_circle:|com/starrocks/planner/SchemaScanNode.java|4|4|100.00%|

**Review by kangkaisen — DISMISSED** [review]

File: fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java [hunk 4]
```diff
@@ -471,6 +476,7 @@ public TDescribeTableResult describeTable(TDescribeTableParams params) throws TE
             return result;
         }
         Database db = Catalog.getCurrentCatalog().getDb(params.db);
+        long limit = params.limit;
```
Comment (stdpain) [inline fe/fe-core/src/main/java/com/starrocks/service/FrontendServiceImpl.java:479]:
```suggestion
        long limit = param.isSetLimit() ? params.getLimit() : -1;
```

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 14 / 20 (70.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/service/FrontendServiceImpl.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/sql/plan/PlanFragmentBuilder.java|10|10|100.00%|
|:large_blue_circle:|com/starrocks/planner/SchemaScanNode.java|4|4|100.00%|

**Review by stdpain — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Comment by kangkaisen** [comment]
for https://github.com/StarRocks/starrocks/issues/4028

**Pull request merged**